src/promptflow/promptflow/_core/token_provider.py

```diff
@@ -41,7 +41,21 @@ def _init_instance(self):
             # Initialize a credential instance
             from azure.identity import DefaultAzureCredential
 
-            self.credential = DefaultAzureCredential()
+            # Support sovereign cloud cases, like mooncake, fairfax.
+            from azure.ai.ml._azure_environments import (
+                _get_default_cloud_name, EndpointURLS, _get_cloud, AzureEnvironments
+            )
+
+            cloud_name = _get_default_cloud_name()
+            if cloud_name != AzureEnvironments.ENV_DEFAULT:
```

Comment:
maybe extract this to utils, I saw _local_azure_connection_operations.py and aoai_gpt4v.py has same code lines.

Reply:
extracted to _utils.credential_utils